File: 213-House-Robber-II.cs
using System;
namespace AlgoPractice
{
	public class Solution
	{
		public int Rob(int[] nums)
		{
			var startFromHouse1 = RobHelper(nums, 0, nums.Length - 2);
			var startFromHouse2 = RobHelper(nums, 1, nums.Length - 1);
			var result = Math.Max(startFromHouse1, startFromHouse2);
			return result;
		}

		public int RobHelper(int[] num, int start, int end)
        {
			var rob1 = 0;
			var rob2 = 0;
			for(var i = start; i<= end; i++)
            {
				var temp = Math.Max(rob1 + num[i], rob2);
				rob1 = rob2;
				rob2 = temp;
            }
			return rob2;
        }
	}
}

File: javascript/213-House-Robber-II.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 var rob = function(nums) {
	const n = nums.length;
	if (n == 1) return nums[0];

	function houseRobber(start, end) {
		let first = 0;
		let second = nums[start];
		for (let i = start +1; i < end; i++) {
			let temp = Math.max(first + nums[i], second);
			first = second;
			second = temp;
		};
		return second;
	};

	return Math.max(houseRobber(0, n-1), houseRobber(1, n));
};